addons/event/tests/test_event_mail_schedule.py

```diff
@@ -341,3 +341,53 @@ def test_archived_event_mail_schedule(self):
 
         # check that scheduler is not executed
         self.assertFalse(event_prev_scheduler.mail_done, 'event: reminder scheduler should should have run')
+
+    @mute_logger('odoo.addons.base.models.ir_model', 'odoo.models')
+    def test_include_mails_to_unconfirmed_attendees_in_the_sent_count(self):
+        """ Emails are sent to both confirmed and unconfirmed attendees.
+        This test checks that the count of sent emails includes the emails sent to unconfirmed ones.
+
+        Time in the test is frozen to simulate the following state:
+
+                   NOW     Event Start    Event End
+                  12:00       13:00        14:00
+                    |           |            |
+            ──────────────────────────────────────►
+            |                   |                time
+            ◄─────────────────►
+                  3 hours
+              Trigger before event
+        """
+        # Deactivate other schedulers to avoid messing with crons
+        self.env['event.mail'].search([]).unlink()
+
+        # Setup
+        now = datetime(2023, 1, 1, 12, 00, 00)
+        event_date_begin = now + timedelta(hours=1)
+        event_date_end = event_date_begin + timedelta(hours=1)
+
+        with freeze_time(now):
+            test_event = self.env['event.event'].with_user(self.user_eventmanager).create({
+                'name': 'TestEventMail',
+                'auto_confirm': True,
+                'date_begin': event_date_begin,
+                'date_end': event_date_end,
+                'event_mail_ids': [
+                    (0, 0, {  # 3 hours before event
+                        'interval_nbr': 3,
+                        'interval_unit': 'hours',
+                        'interval_type': 'before_event',
+                        'template_ref': 'mail.template,%i' % self.env['ir.model.data']._xmlid_to_res_id('event.event_reminder')})
+                ]
+            })
+            reg_draft = self.env['event.registration'].with_user(self.user_eventuser).create({
+                'event_id': test_event.id,
+                'name': 'RegistrationUnconfirmed',
+                'email': 'Registration@Unconfirmed.com',
+            })
+            reg_draft.write({'state': 'draft'})
+
+           # Check event scheduler
+            scheduler = self.env['event.mail'].search([('event_id', '=', test_event.id)])
```

Comment:
`test_event.event_mail_ids` not sure what do you mean?

Reply:
Instead of searching for the scheduler just fetch it with regular python get. Should be the same result of do `test_event.event_mail_ids` or `env['event.mail'].search([('event_id', '=', test_event.id)])`, except using the first one will check the cache first (and likely find it).

Unless there's another reason to do the search?

Reply:
ok, thanks for the explanation